### PR TITLE
update validation for new data + zoom driven functions

### PR DESCRIPTION
**Pull request by ansis**
:eyes: @lucaswoj

### CODE REVIEW

**Comment by lucaswoj** [comment]
Per chat with @scothis, we're going to relax some of these restrictions. Ping me when this PR is ready for 👀.

**Comment by ansis** [comment]
@lucaswoj this should be ready for :eyes:. 7bc1c53 relaxes restrictions. Do any changes need to be made the spec itself?

File: lib/validate/validate_function.js [hunk 1]
```diff
@@ -51,13 +53,29 @@ module.exports = function validateFunction(options) {
...
+        }
+
+        if (type === 'object') {
+            errors = errors.concat(validate({
+                key: key + '[0].zoom',
+                value: value[0].zoom,
+                valueSpec: {type: 'number'},
+                style: options.style,
+                styleSpec: options.styleSpec
+            }));
```
Comment (lucaswoj) [inline lib/validate/validate_function.js:77]: Should we also validate the `value` property, if it is present?

**Comment by lucaswoj** [comment]
> Do any changes need to be made the spec itself?

Nope, you've captured it all in code 👍.

File: lib/validate/validate_function.js (hunk 1 above)
Comment (lucaswoj) [inline lib/validate/validate_function.js:77]: I think we can assert that
- `value` is either a string or a number
- no keys other than `value` and `zoom` are present

**Comment by lucaswoj** [comment]
Was the change to `bin/gl-style-composite` an accident?

**Comment by lucaswoj** [comment]
This is good to 🚢 once you add a test for the new zoom-and-feature function syntax.

**Comment by ansis** [comment]
d45b8734c49dd8df446c94c1bfda3f1320830c9f adds validation tests for zoom+feature functions

**Pull request merged**